editor/src/core/workers/parser-printer/parser-printer-comments.ts

```diff
@@ -0,0 +1,64 @@
+import * as TS from 'typescript'
+import { Comment, multiLineComment, singleLineComment } from '../../shared/element-template'
+import { fastForEach } from '../../shared/utils'
+
+export interface CommentsResult {
+  leadingComments: Array<Comment>
+}
+
+export function getComments(sourceText: string, node: TS.Node): CommentsResult {
```

Comment:
So is the idea then that we could effectively run this against all of the nodes we parse to progressively add more and more comment support?

Reply:
Yes, there's some varied complexity around this as described in a larger comment block, but that should work, also for eventually adding trailing comments.